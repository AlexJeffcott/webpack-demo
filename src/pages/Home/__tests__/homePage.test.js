import React from 'react'
import { cleanup, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import HomePage from '../'

describe(`HomePage`, () => {
  afterEach(cleanup)

  const testData = [
    [
      'has correct simple title text',
      {},
      {
        h1: `Hello React!`,
        titleComponent: `Hello Again React!`,
        p: `hi there`,
      },
    ],
  ]

  test.each(testData)(`%p`, (testTitle, testParams, testExpected) => {
    const { asFragment, getByTestId, getByText } = render(<HomePage />)
    expect(getByTestId('HomePage Component')).toHaveTextContent(testExpected.h1)
    expect(getByText(testExpected.h1)).toBeInTheDocument()

    expect(getByTestId('Title Component')).toHaveTextContent(
      testExpected.titleComponent
    )
    expect(getByText(testExpected.titleComponent)).toBeInTheDocument()

    expect(getByText(testExpected.p)).toBeInTheDocument()

    expect(asFragment()).toMatchSnapshot()
  })
})
