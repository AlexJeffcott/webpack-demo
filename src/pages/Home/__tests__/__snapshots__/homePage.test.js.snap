// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage "has correct simple title text" 1`] = `
<DocumentFragment>
  <div
    data-testid="HomePage Component"
  >
    <h1>
      Hello React!
    </h1>
    <h1
      data-testid="Title Component"
    >
      Hello Again React!
    </h1>
    <p>
      hi there
    </p>
    <img
      alt="logo"
      src="png-stub"
    />
    <svg-stub />
  </div>
</DocumentFragment>
`;
